src/app/ui/email-history-card.tsx

```diff
@@ -42,110 +18,163 @@ interface senderType {
+                      {item.sender_local_part}@aws-educate.tw
+                    </p>
                   </div>
-                  <div className="flex-1 flex py-2 gap-2 items-center">
-                    <div className="bg-sky-800 rounded-full p-1 border-2 border-sky-800">
-                      <Sheet size={32} color="white" />
-                    </div>
-                    <div className="flex flex-col justify-start">
-                      <p className="text-md font-medium text-black">Spreadsheet file</p>
-                      <p className="text-sm font-medium text-neutral-500">
-                        {item.spreadsheet_file.file_name}
-                      </p>
-                    </div>
+                </div>
+
+                {/* Recipient Source Info */}
+                {renderRecipientInfo(item)}
+
+                {/* Template File Info */}
+                <div className="flex-1 flex py-2 gap-2 items-center">
+                  <div className="bg-sky-700 rounded-full p-1 border-2 border-sky-700">
+                    <FileText size={32} color="white" />
                   </div>
-                  <div className="flex-1 flex py-2 gap-2 items-center">
-                    <div className="bg-sky-700 rounded-full p-1 border-2 border-sky-700">
-                      <FileText size={32} color="white" />
-                    </div>
-                    <div className="flex flex-col justify-start">
-                      <p className="text-md font-medium text-black">template file</p>
-                      <p className="text-sm font-medium text-neutral-500">
-                        {item.template_file.file_name}
-                      </p>
-                    </div>
+                  <div className="flex flex-col justify-start">
+                    <p className="text-md font-medium text-black">Template File</p>
+                    <p className="text-sm font-medium text-neutral-500">
+                      {item.template_file.file_name}
+                    </p>
                   </div>
                 </div>
               </div>
-              <div className="flex flex-col justify-end pt-2 pl-3 gap-2 lg:flex-row  lg:pt-8  lg:gap-8">
+
+              {/* Footer Info */}
+              <div className="flex flex-col justify-end pt-2 pl-3 gap-2 lg:flex-row lg:pt-8 lg:gap-8">
                 <div className="flex items-center gap-2">
                   <div className="flex items-center gap-2">
                     <CalendarDays size={20} />
                     <p>Send at</p>
                   </div>
                   <strong>{convertToTaipeiTime(item.created_at)}</strong>
                 </div>
+
                 <div className="flex items-center gap-2">
                   <div className="flex items-center gap-2">
                     <Send size={20} />
                     <p>Sender</p>
                   </div>
                   <strong>{item.sender.username}</strong>
```

Comment:
Deal with the empty Sender 

<img width="1091" alt="image" src="https://github.com/user-attachments/assets/95f1bf49-7ccd-4916-a03c-398c2cd058f1" />
<img width="1090" alt="image" src="https://github.com/user-attachments/assets/8874da6a-8e97-4788-9701-008b460d5a88" />